apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/CustomIdGenerator.kt

```diff
@@ -0,0 +1,24 @@
+package com.apollographql.apollo.compiler
+
+interface CustomIdGenerator {
```

Comment:
nit: should we just call it `OperationIdGenerator` instead? And make `CodeGenerationContext` always have value set to Sha256 ?

Reply:
There was `customTypeMapping`. I followed the same convention. Should I change it to `OperationIdGenerator`? Or 

the interface name as `OperationIdGenerator` and the input name as `customIdGenerator`. 

```
apollo {
  customIdGenerator = new OperationIdGenerator() {
    // ...
  }
}
```